sqlglot/helper.py

```diff
@@ -45,6 +46,25 @@ def __get__(self, obj: t.Any, owner: t.Any = None) -> t.Any:
         return classmethod(self.fget).__get__(None, owner)()  # type: ignore
 
 
+def suggest_closest_match_and_fail(
+    kind: str,
+    word: str,
+    possibilities: t.Iterable[str],
+    warn_only: bool = False,
```

Comment:
why do you need a warn mode?

Reply:
Addressed here: https://github.com/tobymao/sqlglot/pull/5119/commits/9521ebff17bd6d06f9400f6cf1d9d7cf9582acbb